### ISSUE TITLE
re-enable intra-pod and node-pod networking tests

### ISSUE THREAD

**Issue by deads2k** [openshift/origin#17754]
https://github.com/openshift/origin/blob/master/test/extended/util/test.go#L260-L262 were disabled in the rebase along with https://github.com/openshift/origin/issues/17750 and https://github.com/openshift/origin/issues/17752.

They should pass (they passed in 3.8).  The latest thoughts are around pod node selectors.  I'd suggest an approach like https://github.com/openshift/origin/pull/17753 or https://github.com/openshift/origin/pull/17751.

@ironcladlou can you summarize your findings again here?

**Comment by ironcladlou** [comment]
Scheduler logs indicated that the scheduler was aware of 4 nodes during the failing tests, but in each case one of the `netserver-` pods created during the test failed to be scheduled on any of the test nodes due to a MatchNodeSelector predicate failure against all 4 nodes. On the surface I think this indicates the node selector for the pods was somehow mismatched against the nodes cached in the scheduler. Need to reproduce and get a dump of the scheduler's node cache and the pods at the time of the predicate failure.

**Comment by ironcladlou** [comment]
Example scheduler error:

```
Dec 12 17:32:24.566: INFO: At 2017-12-12 17:27:02 +0000 UTC - event for netserver-2: {default-scheduler } FailedScheduling: 0/4 nodes are available: 4 MatchNodeSelector.
```

Also of interest is that scheduling doesn't seem to be retried for the failing pod.

**Comment by danwinship** [comment]
> Also of interest is that scheduling doesn't seem to be retried for the failing pod.

given that it failed because (allegedly) none of the nodes matched its NodeSelector, it makes sense for it to not be retried until a node is added or modified

**Comment by ironcladlou** [comment]
>>Also of interest is that scheduling doesn't seem to be retried for the failing pod.

>given that it failed because (allegedly) none of the nodes matched its NodeSelector, it makes sense for it to not be retried until a node is added or modified

I'm not sure that's how the scheduler works. Pods only appear to be queued internally in response to pod informer events (and resyncs) and unhandled errors (which this particular one is not). Maybe I'm missing something in the code.

**Comment by danwinship** [comment]
OK, so the problem is that the pods are getting `role: app` added to their NodeSelectors, causing them to be unable to deploy to the master node, which is labeled `role: infra`. Presumably that comes from the Namespace `node-selector` annotation, but #17760 doesn't seem to be fixing it.

**Comment by danwinship** [comment]
OK, #17760 doesn't fix it because the Namespace doesn't actually have either of the `node-selector` annotations:

    STEP: removing the node-selector annotations on &v1.Namespace{ ... Annotations:map[string]string{"openshift.io/sa.scc.supplemental-groups":"1000960000/10000", "openshift.io/sa.scc.uid-range":"1000960000/10000", "openshift.io/sa.scc.mcs":"s0:c31,c15"} ... }

(from the latest run of #17759)

**Comment by danwinship** [comment]
so... where else can the amended pod NodeSelector be coming from?

**Comment by danwinship** [comment]
Right, forgot about this bug. So the current status is that Pods are getting `role: app` added to their NodeSelectors, which makes the test fail, but I can't figure out where that label is coming from; the Namespace does not appear to have a `node-selector` annotation. So where else could the `role: app` be coming from?
